internal/controller/build_reconciler.go

```diff
@@ -235,15 +179,79 @@ func (r *BuildReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *BuildReconciler) reconcileUploadFile(ctx context.Context, obj BuildableObject) (result, error) {
+	log := log.FromContext(ctx)
+
+	spec := obj.GetBuild().Upload
+	status := obj.GetStatusBuild()
+
+	if spec.RequestID != status.UploadRequestID {
+		url, expiration, err := r.generateSignedURL(obj)
+		if err != nil {
+			return result{}, fmt.Errorf("generating upload url: %w", err)
+		}
+
+		obj.SetStatusBuild(apiv1.BuildStatus{
+			UploadURL:        url,
+			UploadRequestID:  spec.RequestID,
+			UploadExpiration: metav1.Time{Time: expiration},
+			Uploaded:         false,
+		})
+		meta.SetStatusCondition(obj.GetConditions(), metav1.Condition{
+			Type:               apiv1.ConditionBuilt,
+			Status:             metav1.ConditionFalse,
+			Reason:             apiv1.ReasonAwaitingUpload,
+			ObservedGeneration: obj.GetGeneration(),
+			Message:            "Waiting for object upload to complete",
+		})
+		if err := r.Client.Status().Update(ctx, obj); err != nil {
+			return result{}, fmt.Errorf("updating status: %w", err)
+		}
+	}
+
+	// Verify the object has been uploaded to storage.
+	storageMd5, err := r.storageObjectMd5(obj, r.SCI)
+	if err != nil {
+		return result{}, fmt.Errorf("getting storage object md5: %w", err)
+	}
+	if storageMd5 != spec.MD5Checksum {
+		log.Info("The object's md5 does not match the spec md5. An upload may be in progress.")
+		// Allow the client to trigger a retry (they can update an annotation).
+		return result{Result: ctrl.Result{RequeueAfter: time.Second}}, nil
+	}
+
+	{
+		s := obj.GetStatusBuild()
+		s.Uploaded = true
+		s.UploadURL = ""
+		s.UploadExpiration = metav1.Time{}
+		obj.SetStatusBuild(s)
+	}
```

Comment:
Are the curly braces a typo?

Reply:
Yeah, its not needed... cant decide on whether to use this practice or not.